libraries/AP_HAL_ESP32/boards/esp32s3devkit.h

```diff
@@ -14,166 +14,138 @@
-//#define HAL_DISABLE_ADC_DRIVER 1
-#define TRUE 1
-#define HAL_USE_ADC TRUE
+#define AP_COMPASS_ENABLE 0
+/*
+#define AP_COMPASS_AK8963_ENABLED 1
+#define HAL_COMPASS_DEFAULT HAL_COMPASS_AK8963_MPU9250
 
-// the pin number, the gain/multiplier associated with it, the ardupilot name for the pin in parameter/s.
-//
-// two different pin numbering schemes, both are ok, but only one at a time:
-#define HAL_ESP32_ADC_PINS_OPTION1 {\
-	{ADC1_CHANNEL_4, 11, 1},\
-	{ADC1_CHANNEL_3, 11, 2},\
-	{ADC1_CHANNEL_1, 11, 3},\
-	{ADC1_CHANNEL_0, 11, 4}\
-}
-#define HAL_ESP32_ADC_PINS_OPTION2 {\
-	{ADC1_GPIO35_CHANNEL, 11, 35},\
-	{ADC1_GPIO34_CHANNEL, 11, 34},\
-	{ADC1_GPIO39_CHANNEL, 11, 39},\
-	{ADC1_GPIO36_CHANNEL, 11, 36}\
-}
-// pick one:
-//#define HAL_ESP32_ADC_PINS HAL_ESP32_ADC_PINS_OPTION1
-#define HAL_ESP32_ADC_PINS HAL_ESP32_ADC_PINS_OPTION1
+#define HAL_MAG_PROBE_LIST ADD_BACKEND(DRIVER_AK8963, AP_Compass_AK8963::probe_mpu9250(GET_I2C_DEVICE(0,0x0c), ROTATION_YAW_270));
+*/
 
-#define HAL_INS_MPU9250_NAME "mpu9250"
+#define ALLOW_ARM_NO_COMPASS
+#define AP_AIRSPEED_ENABLED 0
+#define AP_AIRSPEED_ANALOG_ENABLED 0
+#define AP_AIRSPEED_BACKEND_DEFAULT_ENABLED 0
 
-// uncommenting one or more of these will give more console debug in certain areas..
-//#define INSEDEBUG 1
-//#define STORAGEDEBUG 1
-//#define SCHEDDEBUG 1
-//#define FSDEBUG 1
-//#define BUSDEBUG 1
+// no ADC
+//#define HAL_DISABLE_ADC_DRIVER 0
+#define HAL_USE_ADC 1
 
-#define HAL_INS_PROBE_LIST PROBE_IMU_SPI( Invensense, HAL_INS_MPU9250_NAME, ROTATION_NONE)
-//#define HAL_INS_PROBE_LIST PROBE_IMU_SPI( Invensense, HAL_INS_MPU9250_NAME, ROTATION_ROLL_180)
-// MAG/COMPASS probing:
-#define AP_COMPASS_AK8963_ENABLED TRUE
-#define HAL_MAG_PROBE_LIST ADD_BACKEND(DRIVER_AK8963, AP_Compass_AK8963::probe_mpu9250(0, ROTATION_NONE))
-#define HAL_PROBE_EXTERNAL_I2C_COMPASSES 1
+#define HAL_ESP32_ADC_PINS_OPTION1 { \
+    {ADC1_GPIO1_CHANNEL, 11, 1}, \
+    {ADC1_GPIO5_CHANNEL, 11, 2} \
+}
 
-//#define HAL_BARO_PROBE_LIST PROBE_BARO_SPI(BMP280, "bmp280")
+#define HAL_ESP32_ADC_PINS HAL_ESP32_ADC_PINS_OPTION1
 
+// /* -- Disable WiFi
 // 2 use udp, 1 use tcp...  for udp,client needs to connect as UDPCL in missionplanner etc to 192.168.4.1 port 14550
-#define HAL_ESP32_WIFI 1
-
-// tip: if u are ok getting mavlink-over-tcp or mavlink-over-udp and want to disable mavlink-over-serial-usb
-//then set ardupilot parameter SERIAL0_PROTOCOL = 0 and reboot.
-// u also will/may want..
-//LOG_BACKEND_TYPE 1
-//LOG_DISARMED 1
-//SERIAL0_PROTOCOL 0
-
-
-// see boards.py
-#ifndef ENABLE_HEAP
-#define ENABLE_HEAP 1
-#endif
-
+#define HAL_ESP32_WIFI 2
 #define WIFI_SSID "ardupilot123"
 #define WIFI_PWD "ardupilot123"
+// */
 
 //RCOUT which pins are used?
 
-#define HAL_ESP32_RCOUT { GPIO_NUM_11,GPIO_NUM_10, GPIO_NUM_9, GPIO_NUM_8, GPIO_NUM_7, GPIO_NUM_6 }
+#define HAL_ESP32_RCOUT { GPIO_NUM_6, GPIO_NUM_7, GPIO_NUM_10, GPIO_NUM_11, GPIO_NUM_12, GPIO_NUM_13, GPIO_NUM_16 }
```

Comment:
see my above note/s about this sort of thing breaking Nick's hardware. :-)

Reply:
same as above.